### PR TITLE
Update the CHANGELOG.md to reflect the current state of development. …

### PR DESCRIPTION
**Pull request by phaazon**
…[skip ci]

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,59 @@
...
+- Add the possibility to switch off the internal logger of libcore — this rely on the
+  `DynamicObject` that now has a new boolean configuration, `"ENABLE_INTERNAL_LOGGING"`. Set it
+  to `true` to enable the internal logging (default), otherwise, set to `false` to completely
+  disable internal logging (this also will prevent the core library from creating log files).
+
+## 2.3.0
+
+> 2019/01/21
+
+- Change the way the library is compiled and linked by using explicti interface libraries. This
```
Comment (KhalilBellakrid) [inline CHANGELOG.md]: 'explicit' instead of 'expliciti'

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+  also fixes a Windows issues related to non-explicitly exported symbols.
+
+## 2.2.0
+
+> 2019/01/15
+
+- Add a new mechanism to `DatabaseEngine`. This allows for custom database driver, which should
+  allow database switching in easier ways.
```
Comment (KhalilBellakrid) [inline CHANGELOG.md]: IMHO it's better to say : 'This allows custom database driver, which allow switching between database backends easier.'

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,59 @@
...
+
+- Implement `Preferences` encryption via LevelDB. Only the values are encrypted. Be careful
+  though: this change doesn’t provide a solution when you want to reset or change your password
+  (already encrypted values will remain still). A fix for this is foreseen later.
+
+## 2.0.0
+
+> 2019/01/04
+
+- Integration of Ethereum as an officially supported currency and currency family into libcore.
```
Comment (KhalilBellakrid) [inline CHANGELOG.md]: IMHO better to say: 'Support of new coin family : Ethereum Like currencies, with an integration of Ethereum as first currency of this family'
You can also add: 'Support of ERC20 tokens'
and 'We do not support catch  other smart contracts which are not ERC20, to be more specific, internal transactions are not caught during synchronization of ethereum accounts'

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,59 @@
...
-0.9.0 (2018-05-14)
-==================
+### [Next Release Here]
 
-First beta release of libledger-core for Ledger Vault integration
\ No newline at end of file
+> 2019/??/??
+
+- Change the internal representation of `DynamicObject`. The new representation is optimized to
+  take less memory and to be safer in terms of typing. The internal interfaces were also
```
Comment (KhalilBellakrid) [inline CHANGELOG.md]: IMHO be better: '... is optimized not only memory wise but also type safety wise '

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,59 @@
...
+  redesigned to allow for a smoother transition to C++17’s `std::variant` when we see fit.
```
Comment (KhalilBellakrid) [inline CHANGELOG.md]: just 'redesigned to allow a smoother ...'

File: CHANGELOG.md (hunk 1 above)
Comment (phaazon) [reply CHANGELOG.md]: 😆

File: CHANGELOG.md (hunk 2 above)
Comment (phaazon) [reply CHANGELOG.md]: Yep!

**Comment by phaazon** [comment]
Thanks for the reviews guys! I’ll address all that soon! :)

**Pull request merged**